ddataflow/ddataflow.py

```diff
@@ -224,14 +225,32 @@ def source_name(self, name, disable_view_creation=False):
 
     def path(self, path):
         """
-        returns a deterministic path replacing the real production path with one based on the current environment needs
+        returns a deterministic path replacing the real production path with one based on the current environment needs.
+        Currently support path starts with 'dbfs:/' and 's3://'.
         """
         if not self._ddataflow_enabled:
             return path
 
         base_path = self._get_current_environment_data_folder()
 
-        return base_path + "/" + path.replace("dbfs:/", "")
+        for path_prefix in ["dbfs:/", "s3://"]:
+            path = path.replace(path_prefix, "")
+
+        return base_path + "/" + path
+
+    def set_up_database(self, db_name: str, spark):
```

Comment:
should we setup the database by default or people will have to call it explicitly?

Reply:
That's a good question! Since most of the pipelines will set up their own spark (e.g. [search-ranking-pipeline](https://github.com/getyourguide/search-ranking-pipeline/blob/31dc19d9b73c788824a3b185e8da20d3fb408f70/search_ranking_pipeline/__init__.py#L26)) when starting the run, I wonder will this [line](https://github.com/getyourguide/DDataFlow/blob/e9773987ca3613a40966a3de6cb526b281ffea45/ddataflow/utils.py#L4) get the same spark session? If yes, we can then maybe make it as an optional setup from the cofig? Since I don't think every project would need a db.

Reply:
And I wonder do we have any clean-up mechanism for these temporary files or tables? Do you think that would be needed?

Reply:
I assume that we will always get the same spark session in a single job so I updated the code with the _get_spark function. Users just need to provide the db name they are using.